packages/react-storage/src/components/StorageBrowser/actions/handlers/utils.ts

```diff
@@ -50,3 +52,43 @@ export const constructBucket = ({
   bucketName: string;
   region: string;
 } => ({ bucketName, region });
+
+export const parseLocationAccess = (location: LocationAccess): LocationData => {
+  const { permission, scope, type } = location;
+  if (!scope.startsWith('s3://')) {
+    throw new Error(`Invalid scope: ${scope}`);
+  }
+
+  const id = crypto.randomUUID();
+
+  // remove default path
+  const sanitizedScope = scope.slice(5);
+  let bucket, prefix;
+
+  switch (type) {
+    case 'BUCKET': {
+      // { scope: 's3://bucket/*', type: 'BUCKET', },
+      bucket = sanitizedScope.slice(0, -2);
+      prefix = '';
+      break;
+    }
+    case 'PREFIX': {
+      // { scope: 's3://bucket/path/*', type: 'PREFIX', },
+      bucket = sanitizedScope.slice(0, sanitizedScope.indexOf('/'));
```

Comment:
Is it possible for `indexOf` to return `-1`? (and is that desirable?)

Reply:
Good question - we have an open issue to filter out `PREFIX` locations that do not end with a trailing slash which _should_ prevent the possibility of `indexOf` resolving to `-1`. Probably makse sense to address it here tho as well, can address in a follow up